zcash_client_backend/src/data_api/wallet/input_selection.rs

```diff
@@ -769,6 +811,164 @@ impl<DbT: InputSource> InputSelector for GreedyInputSelector<DbT> {
+            .map_err(InputSelectorError::DataSource)?;
+
+        let input_total = spendable_notes.total()?;
+
+        #[cfg(feature = "orchard")]
+        let (spends_sapling, spends_orchard) = (
+            spendable_notes.sapling().len(),
+            spendable_notes.orchard().len(),
+        );
+        #[cfg(not(feature = "orchard"))]
+        let (spends_sapling, spends_orchard) = (spendable_notes.sapling().len(), 0 as usize);
+
+        let fee_required = match recipient
+            .clone()
+            .convert_if_network(params.network_type())?
+        {
+            Address::Sapling(_) => change_strategy.fee_rule().fee_required(
+                params,
+                target_height,
+                [],
+                [],
+                spendable_notes.sapling().len(),
+                1,
+                spends_orchard,
+            ),
+            Address::Transparent(_) => change_strategy.fee_rule().fee_required(
+                params,
+                target_height,
+                [],
+                [P2PKH_STANDARD_OUTPUT_SIZE],
+                spendable_notes.sapling().len(),
+                0,
+                spends_orchard,
+            ),
+            Address::Unified(addr) => {
+                if cfg!(feature = "orchard") && addr.has_orchard() {
+                    change_strategy.fee_rule().fee_required(
+                        params,
+                        target_height,
+                        [],
+                        [],
+                        spends_sapling,
+                        0,
+                        std::cmp::max(spends_orchard, 1),
+                    )
+                } else if addr.has_sapling() {
+                    change_strategy.fee_rule().fee_required(
+                        params,
+                        target_height,
+                        [],
+                        [],
+                        spends_sapling,
+                        1,
+                        spends_orchard,
+                    )
+                } else if addr.has_transparent() {
+                    change_strategy.fee_rule().fee_required(
+                        params,
+                        target_height,
+                        [],
+                        [P2PKH_STANDARD_OUTPUT_SIZE],
+                        spends_sapling,
+                        0,
+                        spends_orchard,
+                    )
+                } else {
+                    unreachable!()
+                }
+            }
+            Address::Tex(_) => change_strategy
+                .fee_rule()
+                .fee_required(
+                    params,
+                    target_height,
+                    [],
+                    [P2PKH_STANDARD_OUTPUT_SIZE],
+                    spendable_notes.sapling().len(),
+                    match spends_sapling > 0 && spends_orchard == 0 {
+                        true => 2,
+                        false => 0,
+                    },
```

Comment:
Why is this written as a match rather than if/else? Also, I don't understand the `spends_orchard == 0` condition here. Finally, the hardcoded value `2` is probably not ideal; elsewhere we use the `BundleType::num_outputs` method to determine padding, for Sapling and Orchard independently. The same should probably be done here.